### PR TITLE
add clusterset v1beta1

### PR DESCRIPTION
**Pull request by elgnay**
The related task: https://github.com/open-cluster-management/backlog/issues/16155
Signed-off-by: Yang Le <yangle@redhat.com>

### CODE REVIEW

File: cluster/v1beta1/doc.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Package v1beta1 contains API Schema definitions for the cluster v1beta1 API group
+// +k8s:deepcopy-gen=package,register
+// +k8s:conversion-gen=open-cluster-management.io/api/cluster
+// +k8s:defaulter-gen=TypeMeta
+// +k8s:openapi-gen=true
+
+// +kubebuilder:validation:Optional
```
Comment (qiujian16) [inline cluster/v1beta1/doc.go:7]: why need this here?

Comment (elgnay) [reply cluster/v1beta1/doc.go:7]: It sets the default validation rule for fields of structs in this package. Other API versions,  including cluster v1/v1alpha1, work and addon, have this setting as well.

**Pull request merged**